internal/controller/issues.go

```diff
@@ -0,0 +1,180 @@
+			Reason:  "IssueHasPR",
+			Message: "Issue has an open PR",
+		}
+	}
+	if !meta.IsStatusConditionPresentAndEqual(issueObject.Status.Conditions, "IssueHasPR", condition.Status) {
+		meta.SetStatusCondition(&issueObject.Status.Conditions, *condition)
+		return true
+	}
+	return false
+}
+
+// fetchAllIssues gets all issues in a repository with retry for rate limits
+func (r *GithubIssueReconciler) fetchAllIssues(ctx context.Context, owner string, repo string) ([]*github.Issue, error) {
+	opt := &github.IssueListByRepoOptions{}
+	maxRetries := 5
+
+	for attempt := 1; attempt <= maxRetries; attempt++ {
+		allIssues, response, err := r.GitHubClient.Issues.ListByRepo(ctx, owner, repo, opt)
+		if err == nil {
+			r.Log.Info("Fetched issues successfully")
+			return allIssues, nil
+		}
+		if response != nil && response.StatusCode == 403 {
+			resetTime := response.Rate.Reset.Time
+			waitDuration := time.Until(resetTime) + time.Second
+			r.Log.Warn(fmt.Sprintf("Rate limit hit. Retrying after %v seconds.", waitDuration.Seconds()))
+			time.Sleep(waitDuration)
+			continue
+		}
+		return nil, fmt.Errorf("error fetching issues: %v", err)
+	}
+	return nil, fmt.Errorf("exceeded retries fetching issues")
+}
+
+// CloseIssue closes the issue on GitHub
+func (r *GithubIssueReconciler) CloseIssue(ctx context.Context, owner string, repo string, gitHubIssue *github.Issue) error {
+	if gitHubIssue == nil {
+		return errors.New("could not find issue in repository")
+	}
+	state := "closed"
+	closedIssueRequest := &github.IssueRequest{State: &state}
+	_, _, err := r.GitHubClient.Issues.Edit(ctx, owner, repo, *gitHubIssue.Number, closedIssueRequest)
+	if err != nil {
+		return fmt.Errorf("failed to close GitHub issue: %v", err)
+	}
+	r.Log.Info(fmt.Sprintf("Closed GitHub issue: %s", gitHubIssue.GetHTMLURL()))
+	return nil
+}
+
+// CreateIssue creates a new issue in the repository
+func (r *GithubIssueReconciler) CreateIssue(ctx context.Context, owner string, repo string, issueObject *issues.GithubIssue) error {
+	newIssue := &github.IssueRequest{Title: &issueObject.Spec.Title, Body: &issueObject.Spec.Description}
+	createdIssue, response, err := r.GitHubClient.Issues.Create(ctx, owner, repo, newIssue)
+	if err != nil {
+		if response != nil {
+			return fmt.Errorf("failed to create issue: %s, %v", response.Status, err)
+		}
+		return fmt.Errorf("failed to create issue: %v", err)
+	}
+	r.Log.Info(fmt.Sprintf("Created GitHub issue: %s", createdIssue.GetHTMLURL()))
+	return nil
+}
+
+// EditIssue edits the description of an existing issue in the repository
+func (r *GithubIssueReconciler) EditIssue(ctx context.Context, owner string, repo string, issueObject *issues.GithubIssue, issueNumber int) error {
+	editIssueRequest := &github.IssueRequest{Body: &issueObject.Spec.Description}
+	_, _, err := r.GitHubClient.Issues.Edit(ctx, owner, repo, issueNumber, editIssueRequest)
+	if err != nil {
+		return fmt.Errorf("failed to edit issue: %v", err)
+	}
```

Comment:
Don't throw away the response

Reply:
Check the API for what is supposed to be returned and make sure it actually returns that...